frontend/src/store/stepState.ts

```diff
@@ -0,0 +1,18 @@
+import { ref } from 'vue';
+import  type { IconSteps } from '@/enum/IconEnum';
+
+export interface IStepInfo {
+    label: string;
+    active: boolean;
+    icon: IconSteps;
+};
+
+export const stepItems = ref<IStepInfo[]>();
```

Comment:
um... I think we might have a better way to implement this stepItems. 

So

1. We should call this grantAccessStepItems, because later on we could have other stepItems for other components, we need to give a name  that can explain itself
2. We should define its default value here, so 
```
const userInfoStep = ...
const userRoleStep = ...
const orgInfoStep = ...

export const grantAccessStepItems = ref<IStepInfo[]>(
 [...userInfoStep],
 [...userRoleStep]
);
```
3. Create an updateStepItems method: 
```
export const setStepItems = (index, changedStepItemsValue: Object<key: value>) => {
     stepItems.value[index] = {...stepItems.value[index], changedStepItemsValue }
 }
```
The above method will still keep the original data for that step, but change whatever needs to change. So in the grantAccess Component, whenever we need to change the grantAccessStepItems, for example, change the userRoleStep, you will call the method in a way like below. The index for the 2nd step is 1 (array index starts from 0). 
```
setStepItems(1, {active: true, icon: checkmark})
```
The label will remain the same, but active and icon will be changed. 
4. Rename the addNewStep method to grantAccessAddNewStep
5. Since we are now using the grantAccessStepItems just for the grantAccess Component. In the PageTitle component, stepItems should be passed as a parameter, because not all components have it 
<img width="1041" alt="Screen Shot 2023-11-08 at 4 11 29 PM" src="https://github.com/bcgov/nr-forests-access-management/assets/23131735/db556825-eefe-44e2-9a4e-4da180059af6">

Reply:
sure, we can chat about it.